src/proxy/http/HttpSessionManager.cc

```diff
@@ -491,6 +494,11 @@ HttpSessionManager::_acquire_session(sockaddr const *ip, CryptoHash const &hostn
       }
     }
   }
+
+  if (acquired) {
+    Metrics::Gauge::decrement(http_rsb.pooled_server_connections);
```

Comment:
Can you help me think through this? It seems like the decrement and increment calls are reversed in this patch, but maybe I'm not thinking through the logic correctly. Why do we decrement the stat when we acquire a session, but increment when we release it?

Reply:
i agree the logic is confusing.  The metric keeps track of the number of sessions waiting in the pool.
* A transaction acquiring the session removes it from the session pool. 
* A transaction releasing the session adds it to the session pool

Reply:
also note i removed the the check for when a session is removed from the pool.  It could instead be an assert because a session is required to be in both the fqdn and ip hash maps.